patina_dxe_core/src/image.rs

```diff
@@ -2196,4 +2196,317 @@ mod tests {
+            crate::GCD.init(48, 16);  // Normal 48-bit address space
+
+            // DON'T call add_memory_space - leave the RBT empty
+            // This means get_closest_idx will return None for ANY address
+
+            // Create a fake PE info with a section at any address
+            let section = goblin::pe::section_table::SectionTable {
+                name: [0; 8],
+                real_name: None,
+                virtual_size: 0x1000,
+                virtual_address: 0x0,  // Section at offset 0 from image_base
+                size_of_raw_data: 0x1000,
+                pointer_to_raw_data: 0,
+                pointer_to_relocations: 0,
+                pointer_to_linenumbers: 0,
+                number_of_relocations: 0,
+                number_of_linenumbers: 0,
+                characteristics: goblin::pe::section_table::IMAGE_SCN_CNT_CODE | goblin::pe::section_table::IMAGE_SCN_MEM_READ | goblin::pe::section_table::IMAGE_SCN_MEM_EXECUTE,
+            };
+
+            let pe_info = super::UefiPeInfo {
+                sections: vec![section],
+                section_alignment: 0x1000,
+                size_of_image: 0x2000,
+                ..Default::default()
+            };
+
+            let mut image_info = empty_image_info();
+            image_info.image_base = 0x1000 as *mut c_void;  // Any address - RBT is empty so lookup will fail
+            image_info.image_size = 0x2000;
+
+            // Manually construct PrivateImageData with minimal required fields
+            let fake_buffer = alloc::alloc::alloc(alloc::alloc::Layout::from_size_align(0x2000, 0x1000).unwrap());
+
+            // Dummy entry point function
+            extern "efiapi" fn dummy_entry(_: *mut c_void, _: *mut efi::SystemTable) -> efi::Status {
+                efi::Status::SUCCESS
+            }
+
+            let private_info = super::PrivateImageData {
+                image_buffer: core::ptr::slice_from_raw_parts_mut(fake_buffer, 0x2000),
+                image_info: Box::new(image_info),
+                hii_resource_section: None,
+                hii_resource_section_base: None,
+                hii_resource_section_num_pages: None,
+                entry_point: dummy_entry,
+                started: false,
+                exit_data: None,
+                image_info_ptr: core::ptr::null_mut(),
+                image_device_path_ptr: core::ptr::null_mut(),
+                pe_info: pe_info.clone(),
+                relocation_data: Vec::new(),
+                image_base_page: 0x1000,
+                image_num_pages: 2,
+            };
+
+            // Call apply_image_memory_protections directly
+            let _result = super::apply_image_memory_protections(&pe_info, &private_info);
+
+            // Should FAIL with NotFound because the GCD RBT is empty (no descriptors),
+            // so get_closest_idx returns None and get_memory_descriptor_for_address returns NotFound
+            #[cfg(not(debug_assertions))]
```

Comment:
I don't think we need to switch back and forth between debug asserts enabled or not (they always will be in test). Instead, you can do a `#[should_panic]` decorator on this function and the test will pass if it panics. That's not the most ideal because any panic will qualify as success, AFAIK, you can't specify what panic should be the right panic.

Now, something we've talked a lot about in the past is disabling debug asserts for tests and I think we should. I'll file an issue.